clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -1460,10 +1460,14 @@ public void handle(OffsetFetchResponse response, RequestFuture<Map<TopicPartitio
                 if (responseError == Errors.COORDINATOR_LOAD_IN_PROGRESS) {
                     // just retry
                     future.raise(responseError);
-                } else if (responseError == Errors.NOT_COORDINATOR) {
+                } else if (responseError == Errors.COORDINATOR_NOT_AVAILABLE ||
+                           responseError == Errors.NOT_COORDINATOR) {
                     // re-discover the coordinator and retry
                     markCoordinatorUnknown(responseError);
                     future.raise(responseError);
+                } else if (responseError.exception() instanceof RetriableException) {
+                    // retry
+                    future.raise(responseError);
```

Comment:
this is interesting. It makes total sense to me, but I remember that when we revisited this for the new consumer, we were not sure if there was maybe a reason behind this behaviour of failing for all unexpected errors on fetch, even if retriable. 

So back then we decided to keep it unchanged in the new consumer, but if we're moving on to allow all retriables to be retried on fetch offsets, I guess we only need align and behave the same in the new consumer (I will file a Jira to review it)

Reply:
For the context, our main goal was to add `COORDINATOR_NOT_AVAILABLE`. Overall, it makes sense to handle retriable errors here too. I wonder if this could have any undesired side effects but I cannot think of any.

@lianetm I agree that we need to align the new consumer if we decide to keep this change here. If we are still undecided, we could also do the change for both consumers in one PR. What do you think?

Reply:
There's a PR open for the new consumer: https://github.com/apache/kafka/pull/16833

Reply:
This will introduce a change in behaviour at the API level for some retriable errors (the ones that were not being handled before) 

Basically, for retriable errors other than COORDINATOR_LOAD_IN_PROGRESS and NOT_COORDINATOR, in the case that they never recover in time, these 2 api calls change a bit in behaviour: 

1. committed: will throw a `TimeoutException`, vs the `KafkaException` it used to throw before this PR (imo this aligns with the API contract, that states KafkaException for unrecoverable and Timeout if it cannot complete in time) 
2. poll: will simply keep retrying and end up returning empty records, vs. throwing a `KafkaException` wrapping the retriable as it used to do before this PR (also aligns with the contract, that states KafkaException for unrecoverable, and "If the timeout expires, an empty record set will be returned"

It's definitely a change in behaviour that will be noticed if those retriable errors happen, but both changes actually look like fixes to me, that align with the current API contract for both calls, and are just handling all retriables in the same way. Thoughts?